controller/jobs.go

```diff
@@ -87,12 +88,19 @@ func (r *JobRepo) Add(job *ct.Job) error {
 		log.Printf("Unable to parse hostID from %q", job.ID)
 		return ErrNotFound
 	}
+	var meta hstore.Hstore
+	if len(job.Meta) > 0 {
+		meta.Map = make(map[string]sql.NullString, len(job.Meta))
+		for k, v := range job.Meta {
+			meta.Map[k] = sql.NullString{String: v, Valid: true}
+		}
+	}
 	// TODO: actually validate
-	err = r.db.QueryRow("INSERT INTO job_cache (job_id, host_id, app_id, release_id, process_type, state) VALUES ($1, $2, $3, $4, $5, $6) RETURNING created_at, updated_at",
-		jobID, hostID, job.AppID, job.ReleaseID, job.Type, job.State).Scan(&job.CreatedAt, &job.UpdatedAt)
+	err = r.db.QueryRow("INSERT INTO job_cache (job_id, host_id, app_id, release_id, process_type, state, meta) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING created_at, updated_at",
+		jobID, hostID, job.AppID, job.ReleaseID, job.Type, job.State, meta).Scan(&job.CreatedAt, &job.UpdatedAt)
 	if e, ok := err.(*pq.Error); ok && e.Code.Name() == "unique_violation" {
-		err = r.db.QueryRow("UPDATE job_cache SET state = $3, updated_at = now() WHERE job_id = $1 AND host_id = $2 RETURNING created_at, updated_at",
-			jobID, hostID, job.State).Scan(&job.CreatedAt, &job.UpdatedAt)
+		err = r.db.QueryRow("UPDATE job_cache SET state = $3, meta = $4, updated_at = now() WHERE job_id = $1 AND host_id = $2 RETURNING created_at, updated_at",
+			jobID, hostID, job.State, meta).Scan(&job.CreatedAt, &job.UpdatedAt)
```

Comment:
The update doesn't need to set the metadata as the only thing that can change is the state.